### PR TITLE
qa: for the latest kclient it will also return EIO

### PR DESCRIPTION
**Pull request by lxbsz**
Fixes: https://tracker.ceph.com/issues/45100
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephfs/test_damage.py [hunk 1]
```diff
@@ -437,11 +436,7 @@ def test_damaged_dentry(self):
         try:
             self.mount_a.stat("subdir/file_to_be_damaged", wait=True)
         except CommandFailedError as e:
-            if isinstance(self.mount_a, FuseMount):
-                self.assertEqual(e.exitstatus, errno.EIO)
-            else:
-                # Kernel client handles this case differently
-                self.assertEqual(e.exitstatus, errno.ENOENT)
+            self.assertEqual(e.exitstatus, errno.EIO)
```
Comment (batrick) [inline qa/tasks/cephfs/test_damage.py:439]: I tried to double-check that this is a new behavior in the `testing` branch of the kernel client but all the tests from the ticket ran with the testing branch.  So we don't have a comparison.

I worry this is a new behavior and tests with the stock kernel will fail with this code change.

**Review by batrick — CHANGES_REQUESTED** [review]

File: qa/tasks/cephfs/test_damage.py (hunk 1 above)
Comment (lxbsz) [reply qa/tasks/cephfs/test_damage.py:439]: Or let's check the both errnos ? If any of them matches will the test case  success ?

Comment (batrick) [reply qa/tasks/cephfs/test_damage.py:439]: Makes sense to me.

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2020-11-05_00:20:13-fs-wip-pdonnell-testing-20201104.220526-distro-basic-smithi/

failures known.

**Pull request merged**